actions/views/browser.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {GenericAction} from 'mattermost-redux/types/actions';
+
+import {Constants, ActionTypes, WindowSizes} from 'utils/constants';
+
+export function emitBrowserWindowResized(): GenericAction {
```

Comment:
Maybe I am misunderstanding, but I don't think this is much more performant then the previous approach since we still send redux updates for every resize. If there is a memoizing selector on the other end, it helps there to prevent a re-render so we get that. But we still have to go though all the redux stuff every time. It's certainly better than the previous approach though. 

I guess I still don't understand why we need this in redux. It feels like CSS and a custom hook that watches for window resize events should be able to solve any problem.

Reply:
Unless I'm missing something, this is only triggering re-renders when the window width crosses one of the 3 window size boundaries and the value stored in redux actually changes andnot every time the window resize event dispatches the action. Both the functional Global Header component and the class-based Sidebar component only get re-rendered when the value stored in Redux changes and not every time the action is dispatch. That's what the console.logs I left in the code are there to demonstrate ... give it a try.

I'm actually not doing this specifically for Global Header but as a replacement for Utils.isMobile(), which is often getting used as if it was state that triggers component re-renders when the window resizes, even though it obviously doesn't. 

I'm on the fence as to whether the JS or CSS approach is better for Global Header. CSS Media Queries are meant for this kind of situation, but not having unnecessary markup if it can be accomplished performantly is also appealing.

Reply:
Regardless of this exploration, I'm suggesting we [continue with the CSS approach](https://github.com/mattermost/mattermost-webapp/pull/8740#issuecomment-921919067) for GH.

Reply:
I think Dean's right. This'll trigger the reducer to run, but since the reducer returns a string, `combineReducers` won't generate a new state object, so Redux won't trigger an update.

Reply:
As for using a hook, I could see that also working as long as it's only React that cares about window size and that it's newer components that rely on it as well. We have enough old random code that needs it that I don't think we can abandon that entirely in favour of hooks yet. This could even be a stepping stone on the way there though

Reply:
I'm going to leave the hook approach to later as a majority of the uses of Utils.isMobile() at the moment are in class-based components. We can absolutely revisit down the line!